source/common/request_info/request_info_impl.h

```diff
@@ -153,6 +154,19 @@ struct RequestInfoImpl : public RequestInfo {
     (*metadata_.mutable_filter_metadata())[name].MergeFrom(value);
   };
 
+  const Router::MetadataMatchCriteria*
+  metadataMatchCriteria(const Router::MetadataMatchCriteria* metadata_match) {
+    // The request's metadata, if present, takes precedence over the route's.
+    const auto& metadata = dynamicMetadata().filter_metadata();
+    const auto filter_it = metadata.find("envoy.lb");
```

Comment:
See my comment about placement of this code, but you should use the `ENVOY_LB` constant from `common/config/well_known_names.h`.